lib/cylc/task_proxy.py

```diff
@@ -120,16 +119,17 @@ def is_timeout_set(self):
         """Return True if timeout is set."""
         return self.timeout is not None
 
-    def next(self):
+    def next(self, no_exhaust=False):
         """Return the next retry delay if there is one, or None otherwise."""
```

Comment:
Doc-string now inaccurate if `no_exhaust` is `True` - repeats the last delay interval indefinitely?